PA_implementation_overview.md

```diff
@@ -0,0 +1,66 @@
+
+### Generate Bids
+Once all Promises have been resolved, and an interest group’s trusted bidding signals has been fetched (if necessary), and reprioritization has been done (again, if necessary), generateBid() is finally invoked.  If the returned ad is not k-anonymous, it’s immediately invoked a second time, restricting the passed in ads to k-anonymous ones.  The perBuyerTimeout, which defaults to 50 milliseconds, is applied independently to each call to generateBid().  Note that it includes running global scripts, in addition to running generateBid() itself.
+
+Once all generateBid() calls that share an executor are complete, Chrome immediately releases the executor handle for that particular generateBid() call to free up resources.
+
+### Reaching Per Buyer Cumulative Timeout
+Once the Per Buyer Cumulative Timeout is reached, all pending work for a bidder that has yet to fully complete and return a result to the auction are dropped - that includes any currently running generateBid() call, generateBid() calls that have completed but whose bids have yet to reach the browser process (which shouldn’t be common), as well as interest groups at any earlier stage.
+
+So this timeout potentially includes executor process startup time, time to load bidding scripts and trusted bidding signals, time to wait for all interest groups to get bidding signals (if “enableBiddingSignalsPrioritization” is true for any of a bidder’s interest group), and, of course, time to generate a bid.  It does not include time waiting for Promises passed into the AuctionConfig to be resolved, time to load interest groups, or time to call scoreAd() for the bids a buyer generates.
+
+Note that if multiple auctions share a bidder process, regardless of whether they share executors or not, they share a Javascript thread, so there is potential contention for a shared resources, while both of their cumulative timeouts run down.  On the other hand, the bidder that starts second may be able to reuse an already created executor, so could avoid having the time to start a process and fetch the bidding script counting against the cumulative timeout.
+
+### Scoring Bids
+Once the seller executor has been loaded, of which there is always only one per AuctonConfig, and Promises resolved, bids are passed to scoreAd() as soon as they’re available, on a first-come-first-served basis.  If any additionalBids are passed in to an auction, they’ll likely be scored first.  Since ties are resolved randomly, and there’s currently no timeouts that span multiple scoreAd() calls, the order in which bids are scored currently does not matter.  Each scoreAd() call independently respects the scoreAd() timeout, which includes the time to run the Javascript file as well as the time to run scoreAd().
```

Comment:
```suggestion
Once the seller executor has been loaded, of which there is always only one per aucton config, and `Promise`s resolved, bids are passed to `scoreAd()` as soon as they’re available, on a first-come-first-served basis.  If any `additionalBids` are passed in to an auction, they’ll likely be scored first.  Since ties are resolved randomly, and there’s currently no timeouts that span multiple `scoreAd()` calls, the order in which bids are scored currently does not matter.  Each `scoreAd()` call independently respects the `scoreAd()` timeout (`sellerTimeout`), which includes the time to run the JavaScript file as well as the time to run `scoreAd()`.
```